pykrige/variogram_models.py

```diff
@@ -36,26 +36,27 @@
 import numpy as np
 
 
-def linear_variogram_model(params, dist):
-    return float(params[0])*dist + float(params[1])
+def linear_variogram_model(params, dist, variogram):
+    variogram[:dist.shape[0], :dist.shape[1]] = (-float(params[0]))*dist - float(params[1])

```

Comment:
Here you assume that `variogram` is a 2d array. That's true for our scado version of uk3d, but not for the other kriging methods in the package. They have an additional empty third dimension. But maybe that's fine for us?

Reply:
Oh, I did not think of this. But... I did notice that I am actually breaking some other calls of the variogram function with this.

I think for us this is fine. But when we want to get PRs merged to the upstream project, we have to get more generic.

Maybe it is a good idea then to show the changes we have to the maintainers and discuss whether they would accept these kinds of things in general. If so, we can spend the work to make it generic. If they do not want to have our performance optimizations anyways, we should probably never bother and just leave it broken!?